### PR TITLE
Add chapter about DMA (with SPI)

### PR DESCRIPTION
**Pull request by bjoernQ**
This adds an introduction to DMA

While we already did some changes regarding DMA for 0.17.0 (and also plan for changes) it's probably good to have something we can adapt

### CODE REVIEW

File: book/src/intro.md
Comment (SergioGasquez) [inline book/src/intro.md:1]: Is there a reason to delete this file?

File: book/src/panic.md
Comment (SergioGasquez) [inline book/src/panic.md:1]: Is there a reason to delete this file?

File: book/src/intro.md
Comment (bjoernQ) [reply book/src/intro.md:1]: Both files seemed redundant to me. Are they used anywhere?

**Review by SergioGasquez — APPROVED** [review]
LGTM!

**Pull request merged**